### ISSUE TITLE
feat(population): Hawkes self-exciting and log-Gaussian Cox process intensities (v1.5)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#115]
## Problem / Request
`point_process.py` fits a homogeneous Gamma-Poisson rate and a log-linear inhomogeneous intensity, and states the LGCP is out of scope for v1. The design's v1.5 rows (`intensity.hawkes`, `intensity.lgcp`, ☐) add the two clustering models real super-emitters need: a Hawkes kernel for event-driven clustering (compressor cycles) and an LGCP with a latent Gauss–Markov process on `log λ` for environment-driven clustering (weather windows). The choice is per-basin by diagnostic.

**Updated after the 2026-09-15 support-repo audits:**
- **Hawkes is already in xtremax** (`xtremax.point_processes.operators.ExponentialHawkes(mu, alpha, beta, observation_window)`, `GeneralHawkesProcess`, `SpatioTemporalHawkes`; exact Ozaki-recursion `log_prob`, `lambda_max`, Ogata sampler, NumPyro wrapper). plumax should not reimplement it — this issue writes the **fit driver, the clustering diagnostic, and the registry entry**. Caveat documented in xtremax `operators/thinning.py:23-30`: composing `ThinningProcess` (detection) over a history-dependent base is an explicit *pseudo*-likelihood, so a POD-thinned Hawkes fit is approximate; say so in the docstring.
- **LGCP is in neither library.** xtremax's own design record delegates GPs to GPJax and that wrapper is unwritten. The first draft of this issue proposed a dense Matérn-3/2 GP via `matern32_covariance` + Cholesky whitening; gaussx (now pinned at `58e90a6`) has no kernel functions but has the state-space Matérn — `gaussx.MaternSDE(variance, lengthscale, order=1)` + `SDEKernel.discretise` + `gaussx.MarkovGaussian` — which gives the latent `log λ` on a time grid as **one NumPyro site with O(T) `log_prob`**. Use that; it is also the same object #103 uses for `Q(t)`, so the two are one parameterisation.

## User Story
> As a user whose basin shows bursty events, I want `fit_hawkes` and `fit_lgcp` alongside the Poisson fits, plus a diagnostic that says which clustering model the data prefer, so the persistency forecasts account for clustering.

## Proposed API
```python
# population/point_process.py
@dataclass(frozen=True)
class HawkesPosterior:  mu: jax.Array; alpha: jax.Array; beta: jax.Array
def fit_hawkes(times, *, horizon, num_samples=1000, ...) -> HawkesPosterior: ...
    # NumPyro model: priors on (mu, alpha, beta) → xtremax ExponentialHawkes(...).log_prob(times, mask)

@dataclass(frozen=True)
class LGCPPosterior:    log_lambda: Float[Array, "S T"]; lengthscale: jax.Array; variance: jax.Array
def fit_lgcp(times, *, grid: Float[Array, "T"], order: int = 1, num_samples=1000) -> LGCPPosterior: ...
    # latent = numpyro.sample("log_lambda", gaussx.MarkovGaussian(A, Q, mu0, P0)) from MaternSDE(order).discretise(Δt)
    # likelihood: xtremax PiecewiseConstantLogIntensity(grid, log_rates=latent) → InhomogeneousPoissonProcess.log_prob (exact .integrate)

def clustering_diagnostic(times, *, horizon) -> dict: ...   # dispersion index, xtremax time_rescaling_residuals + ks_statistic_exp1 per model, ELPD Poisson vs Hawkes vs LGCP
```

## Implementation Steps
- [ ] `fit_hawkes` as a NumPyro model over `xtremax.point_processes.ExponentialHawkes.log_prob`; register `"hawkes"` in the intensity registry from #106 (conditional on history)
- [ ] `fit_lgcp`: `MaternSDE(order)` → `discretise` on the grid spacing → `MarkovGaussian` latent → `PiecewiseConstantLogIntensity` likelihood; expose `order` (1 = Matérn-3/2 default, 0 = OU)
- [ ] `clustering_diagnostic` with the branch-selection guidance from `06_tier5_population.md` / `06b` in the docstring
- [ ] Persistency metrics (#112) accept both posteriors (Hawkes metrics conditional on history)

## Definition of Done
- [ ] Hawkes: synthetic `(μ, α, β)` from `ExponentialHawkes.sample` recovered within posterior σ; `α → 0` reduces to the Gamma-Poisson posterior
- [ ] LGCP: synthetic Matérn-3/2 log-intensity recovered with 95 % CI coverage ≈ 95 %; `MarkovGaussian.covariance_matrix` matches the dense Matérn-3/2 kernel on the grid to 1e-8
- [ ] Likelihood gradients match finite differences
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/population/test_hawkes.py::test_recovery` (`slow`)
- [ ] `tests/population/test_lgcp.py::test_coverage` (`slow`)
- [ ] `tests/population/test_lgcp.py::test_state_space_matches_dense_matern32`
- [ ] `tests/population/test_point_process.py::test_hawkes_alpha_zero_is_poisson`

## Relationships
- Parent (theme epic): #75
- Blocked by: #106 (xtremax pin + intensity registry)
- Related: #103 (same `MaternSDE` / `MarkovGaussian` parameterisation for `Q(t)`), #72 / #91 (spatial Matérn source prior stays dense + Kronecker in plumax)
